### PR TITLE
CustomBarsOverlayFeature: Move parsing to WynnBossBarUtils

### PR DESCRIPTION
**Pull request by kristofbolyai**
As requested by @magicus

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java [hunk 1]
```diff
@@ -301,22 +294,14 @@ public ManaBankBarOverlay() {
...
-                    manaBankMaxPercent = matcher.group(2);
-                    break;
-                }
-            }
+            Pair<LerpingBossEvent, Matcher> matcherPair = WynnBossBarUtils.getLerpingBossEvent(MANA_BANK_PATTERN);
+            LerpingBossEvent bankEvent = matcherPair.a;
 
             if (bankEvent == null) return;
 
+            String manaBankPercent = matcherPair.b.group(1);
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java]: Can you like make a public method in WynnBossBarUtils like `getManaBank` and `getBloodPool(Percent)`? And move the patterns over there. For efficiency, let `getManaBank` return a Pair of strings. (From an API point of view, two methods returning the current and max value would have been nicest, but that'd require doubling the parsing.

File: common/src/main/java/com/wynntils/wynn/utils/WynnBossBarUtils.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.Pair;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.components.LerpingBossEvent;
+
+public class WynnBossBarUtils {
+
+    public static Pair<LerpingBossEvent, Matcher> getLerpingBossEvent(Pattern titlePattern) {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/wynn/utils/WynnBossBarUtils.java]: ... and then this can just be a private helper method.

**Comment by byBackfish** [comment]
We could also just make a `registerPattern` method that accepts a pattern and a callback, and then listen to the BossHealthUpdateEvent (event that is fired once the bossbar updates) and then check for the patterns and fire the callback with the regex match

**Comment by kristofbolyai** [comment]
> We could also just make a `registerPattern` method that accepts a pattern and a callback, and then listen to the BossHealthUpdateEvent (event that is fired once the bossbar updates) and then check for the patterns and fire the callback with the regex match 

I don't know where we would put it, but no need to overcomplicate it imo

**Comment by byBackfish** [comment]
> I don't know where we would put it, but no need to overcomplicate it imo

Would be pretty cool to have something like that, but yea, it might be a bit "overkill"

**Review by magicus — APPROVED** [review]

**Pull request merged**